python/rmm/_lib/lib.pyx

```diff
@@ -119,6 +118,17 @@ def rmm_finalize():
     return 0
 
 
+cdef void _rmmFinalizeWrapper ():
+    global opts
+    free(opts)
+    opts = NULL
+    rmmFinalize()
+
+
+def register_finalize():
+    atexit(&_rmmFinalizeWrapper)
```

Comment:
Can we move this logic to the librmm level and then just have the cython function call the librmm function?

Reply:
I think it's a good idea to keep this in Cython if we need to do other Cython-level cleanup in the future.

Reply:
It's a shame other languages will have to mirror this behavior, but I agree.